spring-cloud-gcp-samples/spring-cloud-gcp-vision-ocr-sample/src/main/java/com/example/WebController.java

```diff
@@ -0,0 +1,142 @@
+public class WebController {
+
+	@Value("${application.ocr-bucket}")
+	private String ocrBucket;
+
+	@Autowired
+	private Storage storage;
+
+	@Autowired
+	private ResourceLoader resourceLoader;
+
+	@Autowired
+	private DocumentOcrTemplate documentOcrTemplate;
+
+	private final OcrStatusReporter ocrStatusReporter;
+
+	public WebController() {
+		this.ocrStatusReporter = new OcrStatusReporter();
+	}
+
+	@GetMapping("/")
+	public ModelAndView renderIndex(ModelMap map) {
+		map.put("ocrBucket", ocrBucket);
+		return new ModelAndView("index", map);
+	}
+
+	@GetMapping("/status")
+	public ModelAndView renderStatusPage(ModelMap map) {
+		map.put("ocrStatuses", ocrStatusReporter.getDocumentOcrStatuses().values());
+		return new ModelAndView("status", map);
+	}
+
+	@GetMapping("/viewDocument")
+	public ModelAndView renderViewDocumentPage(
+			@RequestParam("gcsDocumentUrl") String gcsDocumentUrl,
+			@RequestParam("pageNumber") int pageNumber,
+			ModelMap map)
+			throws ExecutionException, InterruptedException, InvalidProtocolBufferException {
+
+		TextAnnotation textAnnotation =
+				ocrStatusReporter.getDocumentOcrStatuses()
+						.get(gcsDocumentUrl)
+						.getResultSet()
+						.getPage(pageNumber);
+
+		String[] firstWordsTokens = textAnnotation.getText().split(" ", 50);
+
+		map.put("pageNumber", pageNumber);
+		map.put("gcsDocumentUrl", gcsDocumentUrl);
+		map.put("text", String.join(" ", firstWordsTokens));
+
+		return new ModelAndView("viewDocument", map);
+	}
+
+	@PostMapping("/submitDocument")
+	public ModelAndView submitDocument(@RequestParam("documentUrl") String documentUrl) throws IOException {
+
+		// Uploads the document to the GCS bucket
+		Resource documentResource = resourceLoader.getResource(documentUrl);
+		BlobId outputBlobId = BlobId.of(ocrBucket, documentResource.getFilename());
+		BlobInfo blobInfo =
+				BlobInfo.newBuilder(outputBlobId)
+						.setContentType(getFileType(documentResource))
+						.build();
+
+		try (WriteChannel writer = storage.writer(blobInfo)) {
+			ByteStreams.copy(documentResource.getInputStream(), Channels.newOutputStream(writer));
+		}
+
+		// Run OCR on the document
+		GoogleStorageLocation documentLocation =
+				GoogleStorageLocation.forFile(outputBlobId.getBucket(), outputBlobId.getName());
+
+		GoogleStorageLocation outputLocation = GoogleStorageLocation.forFolder(
+				outputBlobId.getBucket(), "ocr_results/" + documentLocation.getBlobName());
+
+		ListenableFuture<DocumentOcrResultSet> result =
+				documentOcrTemplate.runOcrForDocument(documentLocation, outputLocation);
+
+		ocrStatusReporter.registerFuture(documentLocation.uriString(), result);
+
+		return new ModelAndView("submit_done");
+	}
+
+	private static String getFileType(Resource documentResource) {
+		int extensionIdx = documentResource.getFilename().lastIndexOf(".");
+		String fileType = documentResource.getFilename().substring(extensionIdx);
+
+		switch (fileType) {
+			case ".tif":
+				return "image/tiff";
+			default:
```

Comment:
what if it's some random file extension? Will it default to PDF?

Reply:
Done. I made it throw an exception now if a user enters a file type other than PDF or TIFF.